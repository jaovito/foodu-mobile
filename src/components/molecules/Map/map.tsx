import React from 'react';
import {MapProps} from './map.props';
import {Container, MapboxContainer, MapPin} from './map.styles';

export const Map = ({location, onRegionChange}: MapProps) => {
  return (
    <Container>
      <MapboxContainer
        initialRegion={{
          latitude: location.latitude ?? 37.78825,
          longitude: location.longitude ?? -122.4324,
          latitudeDelta: 0.005,
          longitudeDelta: 0.005,
        }}
        onRegionChange={onRegionChange}>
        <MapPin color="red" />
      </MapboxContainer>

      <MapPin color="red" />
    </Container>
  );
};
