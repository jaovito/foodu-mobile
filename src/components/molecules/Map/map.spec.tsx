import {fireEvent, render} from '@test-utils';
import {mock} from '@test/mocks';
import {Map} from './map';

jest.mock('react-native-maps', () => {
  const {View} = require('react-native');
  const MockMapView = (props: any) => {
    return <View>{props.children}</View>;
  };
  const MockMarker = (props: any) => {
    return <View>{props.children}</View>;
  };
  return {
    __esModule: true,
    default: MockMapView,
    Marker: MockMarker,
  };
});

describe('HeaderImage tests', () => {
  it('should be defined', () => {
    const map = render(
      <Map
        location={mock.map.location}
        onRegionChange={mock.map.onRegionChange}
      />,
    );

    expect(map).toBeDefined();
  });

  it('should be able to move the map', () => {
    const map = render(
      <Map
        location={mock.map.location}
        onRegionChange={mock.map.onRegionChange}
      />,
    ).container;

    fireEvent.scroll(map);

    expect(map).toBeDefined();
  });

  it('should match with snapshot', () => {
    const map = render(
      <Map
        location={mock.map.location}
        onRegionChange={mock.map.onRegionChange}
      />,
    ).toJSON();

    expect(map).toMatchSnapshot();
  });

  it('should match snapshot without onchange ', () => {
    const map = render(
      <Map
        onRegionChange={mock.map.onRegionChange}
        location={mock.map.location}
      />,
    ).toJSON();

    expect(map).toMatchSnapshot();
  });
});
