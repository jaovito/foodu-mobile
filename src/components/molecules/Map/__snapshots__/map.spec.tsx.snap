// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderImage tests should match snapshot without onchange  1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "borderColor": "#FFFFFF",
        "borderRadius": 31,
        "borderStyle": "solid",
        "borderWidth": 3,
        "height": "85%",
        "justifyContent": "center",
        "width": "100%",
      },
    ]
  }
>
  <View>
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "red",
            "fontSize": 12,
          },
          [
            {
              "fontSize": 90,
              "position": "absolute",
              "zIndex": 5,
            },
          ],
          {
            "fontFamily": "FontAwesome5Free-Solid",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {
            "fontWeight": "900",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "red",
          "fontSize": 12,
        },
        [
          {
            "fontSize": 90,
            "position": "absolute",
            "zIndex": 5,
          },
        ],
        {
          "fontFamily": "FontAwesome5Free-Solid",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {
          "fontWeight": "900",
        },
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`HeaderImage tests should match with snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "borderColor": "#FFFFFF",
        "borderRadius": 31,
        "borderStyle": "solid",
        "borderWidth": 3,
        "height": "85%",
        "justifyContent": "center",
        "width": "100%",
      },
    ]
  }
>
  <View>
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "red",
            "fontSize": 12,
          },
          [
            {
              "fontSize": 90,
              "position": "absolute",
              "zIndex": 5,
            },
          ],
          {
            "fontFamily": "FontAwesome5Free-Solid",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {
            "fontWeight": "900",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "red",
          "fontSize": 12,
        },
        [
          {
            "fontSize": 90,
            "position": "absolute",
            "zIndex": 5,
          },
        ],
        {
          "fontFamily": "FontAwesome5Free-Solid",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {
          "fontWeight": "900",
        },
      ]
    }
  >
    
  </Text>
</View>
`;
